test/2.0/csv.spec.ts

```diff
@@ -0,0 +1,321 @@
+  t.is(
+    duplicateResult.errors[0].message,
+    "Column hospital_location duplicated in header"
+  )
+  t.deepEqual(duplicateResult.columns, VALID_HEADER_COLUMNS)
+})
+
+test("validateHeaderRow", (t) => {
+  t.is(
+    validateHeaderRow(VALID_HEADER_COLUMNS, []).length,
+    VALID_HEADER_COLUMNS.length
+  )
+  t.is(
+    validateHeaderRow(VALID_HEADER_COLUMNS, [
+      "name",
+      "2022-01-01",
+      "1.0.0",
+      "Woodlawn",
+      "123 Address",
+      "001 | MD",
+      "true",
+    ]).length,
+    0
+  )
+  // last_updated_on must be a valid date
+  const invalidDateResult = validateHeaderRow(VALID_HEADER_COLUMNS, [
+    "name",
+    "2022-14-01",
+    "1.0.0",
+    "Woodlawn",
+    "123 Address",
+    "001 | MD",
+    "true",
+  ])
+  t.is(invalidDateResult.length, 1)
+  t.assert(invalidDateResult[0].message.includes("not a valid YYYY-MM-DD date"))
+  // affirmation must be true
+  const wrongAffirmationResult = validateHeaderRow(VALID_HEADER_COLUMNS, [
+    "name",
+    "2022-01-01",
+    "1.0.0",
+    "Woodlawn",
+    "123 Address",
+    "001 | MD",
+    "yes",
+  ])
+  t.is(wrongAffirmationResult.length, 1)
+  t.assert(wrongAffirmationResult[0].message.includes("allowed value"))
+})
+
+test("validateColumns tall", (t) => {
+  const columns = [
+    ...BASE_COLUMNS,
+    ...TALL_COLUMNS,
+    "code | 1",
+    "code | 1 | type",
+  ]
+  t.is(validateColumns(columns).length, 0)
+  // any order is okay
+  const reverseColumns = [...columns].reverse()
+  t.is(validateColumns(reverseColumns).length, 0)
+  // extra code columns may appear
+  const extraCodes = [
+    ...columns,
+    "code|2",
+    "code|2|type",
+    "code|3",
+    "code|3|type",
+  ]
+  t.is(validateColumns(extraCodes).length, 0)
+  // if a column is missing, that's an error
+  const missingBase = columns.slice(1)
+  const missingBaseResult = validateColumns(missingBase)
+  t.is(missingBaseResult.length, 1)
+  t.assert(missingBaseResult[0].message.includes("description is missing"))
+  // this also applies to code columns, where code|i means that code|i|type must appear
+  const missingCode = [...columns, "code | 2"]
+  const missingCodeResult = validateColumns(missingCode)
+  t.is(missingCodeResult.length, 1)
+  t.assert(missingCodeResult[0].message.includes("code | 2 | type is missing"))
+  // code|i|type means that code|i must be present
+  const missingType = [...columns, "code | 2 | type"]
+  const missingTypeResult = validateColumns(missingType)
+  t.is(missingTypeResult.length, 1)
+  t.assert(missingTypeResult[0].message.includes("code | 2 is missing"))
+})
+
+test("validateRow tall", (t) => {
+  const columns = [
+    ...BASE_COLUMNS,
+    ...TALL_COLUMNS,
+    "code | 1",
+    "code | 1 | type",
+    "code | 2",
+    "code | 2 | type",
+  ]
+  const basicRow = {
+    description: "basic description",
+    setting: "inpatient",
+    "code | 1": "12345",
+    "code | 1 | type": "DRG",
+    "code | 2": "",
+    "code | 2 | type": "",
```

Comment:
@mint-thompson Just to confirm (and I don't want this to hold up this PR), but if you have a "code | 3" _without_ a corresponding "code | 3 | type", the validator would not flag that, correct?

I tired adding a "code | 3" without the corresponding "type" header and reran the tests and they all passed. Just want to confirm current expected behavior.